Library/Homebrew/download_strategy.rb

```diff
@@ -323,7 +342,13 @@ def _fetch
 # Query parameters on the URL are converted into POST parameters
 class CurlPostDownloadStrategy < CurlDownloadStrategy
   def _fetch
-    base_url, data = @url.split("?")
+    base_url, data = if meta.key?(:data)
+      escape_data = ->(k, v) { ["-d", "#{CGI.escape(k.to_s)}=#{CGI.escape(v.to_s)}"] }
```

Comment:
Can the `"#{CGI.escape(k.to_s)}=#{CGI.escape(v.to_s)}"` bit maybe be a shared function where it's used above too?

Reply:
I'd say let's go with the rule of three on this one.

Reply:
Made me think, that there should be something like this for `URI`. And there is: `URI.encode_www_form`

Reply:
Nice!